dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java

```diff
@@ -154,7 +155,13 @@ public int code() {
 
     private static HttpRequest.BodyPublisher toBody(Request request) {
         if (request.body().isPresent()) {
-            return HttpRequest.BodyPublishers.ofInputStream(request.body().get()::content);
+            ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+            try {
+                request.body().get().writeTo(bytes);
+            } catch (IOException e) {
+                throw new SafeRuntimeException("Failed to create a BodyPublisher", e);
+            }
```

Comment:
This is the same block from ConjureBodySerDe, but it's only required for HttpChannel, not socket based transports.